### ISSUE TITLE
Implement settings v5 schema [2d]

### ISSUE THREAD

**Issue by jandubois** [rancher-sandbox/rancher-desktop#3746]
The v5 structure layout is being discussed in #3740 and the final version should be copied over into this story, once the discussion is settled.

This story is basically just a bunch of variable renames.

Please put the move from `kubernetes.containerEngine` to `containerEngine.Name` into a separate commit, because only a subset of the `containerEngine` references are affected, so this will be easier to review.

# Acceptance Criteria

| Scenario | Expected Output | 
| :---: | :---: |
|  Fresh install Rancher Desktop | check schema of created settings.json  |

**Comment by ericpromislow** [comment]
Note on the new option flags for rdctl:

- Names like `kubernetes.memoryInGB` are now being upgraded to the v5 name `virtual-machine.memory-in-gb` to map to the internal JSON fully-qualified field `virtualMachine.memoryInGB`.

- The settings are used in the following contexts like so:

`rdctl start|set ... --virtual-machine.memory-in-gb=NUMBER … `

`rdctl api -X PUT /settings -b '{ "virtualMachine": { "memoryInGB": NUMBER }}'

But on the old direct command-line:
```
open -a /Applications/Rancher\ Desktop.app --virtualMachine.memoryInGB=NUMBER
```

Similar for Windows and Linux

**Comment by ericpromislow** [comment]
See the modifications section in #3740 for the full list of changes to be done.

**Comment by ericpromislow** [comment]
Fixed by #3799
